tidy3d/plugins/mode/mode_solver.py

```diff
@@ -642,14 +643,16 @@ def _has_fully_anisotropic_media(self) -> bool:
 
     @cached_property
     def _has_complex_eps(self) -> bool:
-        """Check if there are media with a complex-valued epsilon in the plane of the mode at the
-        mode solver freqs. A separate check is done inside the solver, which looks at the actual
+        """Check if there are media with a complex-valued epsilon in the plane of the mode.
+        A separate check is done inside the solver, which looks at the actual
         eps and mu and uses a tolerance to determine whether to use real or complex fields, so
         the actual behavior may differ from what's predicted by this property."""
+        check_freqs = np.unique([np.amin(self.freqs), np.amax(self.freqs), np.mean(self.freqs)])
```

Comment:
Just one last minor points: if len(self.freqs) is 1 or 2, check_freqs can be just 1 or 2 as well.

Reply:
There's a `np.unique` that should handle that?

Reply:
Ah yeah